src/modules/hip/kernel/lens_correction.hpp

```diff
@@ -0,0 +1,119 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+// Compute Inverse matrix (3x3)
+__global__ void get_inverse_hip(d_float9 *matTensor, d_float9 *invMatTensor)
+{
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+    d_float9 *mat = &matTensor[id_z];
+    d_float9 *invMat = &invMatTensor[id_z];
+    float det = mat->f1[0] * (mat->f1[4] * mat->f1[8] - mat->f1[7] * mat->f1[5]) - mat->f1[1] * (mat->f1[3] * mat->f1[8] - mat->f1[5] * mat->f1[6]) + mat->f1[2] * (mat->f1[3] * mat->f1[7] - mat->f1[4] * mat->f1[6]);
+    if(det != 0)
+    {
+        float invDet = 1 / det;
+        invMat->f1[0] = (mat->f1[4] * mat->f1[8] - mat->f1[7] * mat->f1[5]) * invDet;
+        invMat->f1[1] = (mat->f1[2] * mat->f1[7] - mat->f1[1] * mat->f1[8]) * invDet;
+        invMat->f1[2] = (mat->f1[1] * mat->f1[5] - mat->f1[2] * mat->f1[4]) * invDet;
```

Comment:
This is long form determinant. Are their faster approaches, pls check

Reply:
@r-abishek 
we are computing here the inverse matrix values

The determinant  is computed in single line here
https://github.com/snehaa8/rpp/blob/sn/lens_correction/src/modules/cpu/kernel/lens_correction.hpp#L33

I have checked online and this is the fastest approach i have seen for computing inverse of a 3x3 matrix